src/Spectre.Data/Datasets/RoiDictionary.cs

```diff
@@ -0,0 +1,110 @@
+﻿/*
+    * RoiDictionary.cs
+    * Class representing a set of tool to manage ROIs.
+    * Allows getting ROIs by name, adding and removing by name.
+
+    Copyright 2017 Roman LIsak
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Spectre.Data.RoiIo;
+
+namespace Spectre.Data.Datasets
+{
+    /// <summary>
+    /// Class representing a set of tools to manage ROIs.
+    /// </summary>
+    /// <seealso cref="Spectre.Data.Datasets.IRoiDictionary" />
+    public class RoiDictionary : IRoiDictionary
+    {
+        private List<Roi> _roiDataset;
+        private RoiReader _roireader = new RoiReader();
+        private string _path;
+
+        /// <summary>
+        /// Initializes the dictionary with specified path
+        /// and corresponding set of ROIs.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        public void InitializeDictionary(string path)
+        {
+            _path = path;
+            _roiDataset = _roireader.GetAllRoisFromDirectory(path);
+        }
+
+        /// <summary>
+        /// Tries to get specified value from list by name.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <returns>
+        /// Returns roi with specified name.
+        /// Returns null if no roi with specified name found.
+        /// </returns>
+        public Roi GetRoiOrDefault(string name)
```

Comment:
Can this function be simplified like:

`public Roi GetRoiOrDefault(string name) => _roiDataset.FirstOrDefault(r => r.Name ==name);`

I think it would suffice for doing exactly the same :)